app/src/lib/stores/app-store.ts

```diff
@@ -2045,12 +2077,126 @@ export class AppStore extends TypedBaseStore<IAppState> {
     const workingDirectory = WorkingDirectoryStatus.fromFiles(updatedFiles)
 
     this.repositoryStateCache.updateChangesState(repository, () => ({
-      diff,
+      selection: {
+        ...changesState.selection,
+        diff,
+      },
       workingDirectory,
     }))
     this.emitUpdate()
   }
 
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _selectStashedFile(
+    repository: Repository,
+    file?: CommittedFileChange | null
```

Comment:
💭 are we able to just use `CommittedFileChange | null` here and omit the `?`? Or if the `undefined` case is unavoidable can we document why it's important here and how it differs from `null`?

Reply:
Right, I made the effort to document this for the working directory case, I just forgot about it here. Let me fix that.

Reply:
Added some more documentation to address this and explain the difference between undefined and null